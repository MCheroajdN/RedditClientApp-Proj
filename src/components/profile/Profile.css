ul.profile-menu {
	list-style: none;
	padding: 0.4rem;
	width: 3rem;
	height: 2rem;
	margin-left: 0.4rem;
	box-shadow: 1px 1px 1px 1px #eeebeb, inset 1px 1px 1px 1px #eeebeb;
	border-radius: 2px;
	text-align: left;
}
ul li {
	position: relative;
}

ul li span {
	color: #353131;
	opacity: 0.6;
	cursor: pointer;
}
ul li ul {
	list-style: none;
	position: absolute;
	top: 40px;
	left: -7rem;
	width: 10rem;
	background-color: #fff;
	text-align: left;
	padding: 1rem 0.4rem 1rem 0.2rem;
	box-shadow: 1px 1px 1px 1px #fff, inset 1px 1px 1px 1px #fff;
	z-index: 2;
}

ul li ul li {
	padding: 0.7rem 0.4rem;
	font-size: 15px;
}

.menu-list a {
	color: #353131;
	text-decoration: none;
}

.menu-list a:hover {
	background-color: #0088ff;
	color: #fff !important;
	transition: all 0.2s ease-in;
	padding: 0.7rem 0.7rem 0.7rem 0;
}
