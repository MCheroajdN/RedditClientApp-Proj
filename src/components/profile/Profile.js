import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import Routes from '../../app/routes';
import Icons from '../../icons';
import './Profile.css';

const Profile = () => {
	const [isMenuList, setIsMenuList] = useState(false);
	const profileDropdown = () => setIsMenuList(!isMenuList);

	const { redditCoins, redditPremium, redditHelp } = Routes;
	return (
		<ul className='profile-menu' onClick={profileDropdown}>
			<li>
				<span aria-label='profile'>
					{Icons.user} {Icons.caretDown}
				</span>
				{isMenuList ? (
					<ul className='menu-list'>
						<small className='p-1'>VIEW OPTIONS</small>
						<li>
							<Link to='/'> {Icons.moon} Night mode </Link>
						</li>
						<small className='p-1'>MORE STUFF</small>
						<li>
							<a href={redditCoins()}>{Icons.copyright} Reddit Coins</a>
						</li>
						<li>
							<a href={redditPremium()}>{Icons.drawPolygon} Reddit Premium</a>
						</li>
						<li>
							<a href={redditHelp()}> {Icons.questionCircle} Help Center</a>
						</li>
						<li>
							<Link
								to='/'
								data-toggle='modal'
								data-target='#loginFormModal'
								role='button'>
								{Icons.arrowRight} Login/Sign up
							</Link>
						</li>
					</ul>
				) : (
					''
				)}
			</li>
		</ul>
	);
};

export default Profile;
