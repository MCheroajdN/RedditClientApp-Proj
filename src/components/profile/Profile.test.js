import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import Profile from './Profile';
import renderer from 'react-test-renderer';

describe('Render <Profile />', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		render(<Profile />, div);
	});
});

describe('Profile Component', () => {
	it('checks state change', () => {
		const profileDropdown = jest.fn();
		const wrapper = shallow(<Profile onClick={profileDropdown} />);

		const handleClick = jest.spyOn(React, 'useState');

		handleClick.mockImplementation((isMenuList) => [
			isMenuList,
			profileDropdown,
		]);

		wrapper.find('.profile-menu').simulate('click');
		expect(profileDropdown).toBeTruthy();
	});
	it('matches the snapshot', () => {
		const tree = renderer.create(<Profile />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
