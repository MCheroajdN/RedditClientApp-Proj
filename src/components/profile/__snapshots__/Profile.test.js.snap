// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component shallow rendering 1`] = `
<ul
  className="profile-menu"
  onClick={[Function]}
>
  <li>
    <span
      role="profile"
    >
      <i
        className="fas fa-user"
      />
       
      <i
        className="fas fa-caret-down"
      />
    </span>
    
  </li>
</ul>
`;
